src/main/java/com/odysseusinc/arachne/portal/api/v1/controller/BaseStudyController.java

```diff
@@ -481,9 +481,11 @@ public void downloadAllFiles(
             @PathVariable("studyId") Long studyId,
             HttpServletResponse response) throws PermissionDeniedException, NotExistException, IOException {
 
-        String archiveName = "study_" + studyId + "_"
-                + Long.toString(System.currentTimeMillis())
-                + ".zip";
+        final Study study = studyService.getById(studyId);
+
+        // replaceAll("\\W+", "_") to get filesystem-friendly name
+        String archiveName = study.getTitle().replaceAll("\\W+", "_") + ".zip";
```

Comment:
Study can contain characters that is not valid to use in the file name
There's utility to sanitize illegal characters from strings:
https://github.com/OHDSI/ArachneCommons/blob/develop/arachne-common-utils/src/main/java/com/odysseusinc/arachne/commons/utils/CommonFilenameUtils.java